buildSrc/src/main/groovy/org/apache/beam/gradle/GrpcVendoring_1_36_0.groovy

```diff
@@ -0,0 +1,223 @@
+
+  static def guava_version = "30.1-jre"
+  static def protobuf_version = "3.15.3"
+  static def grpc_version = "1.36.0"
+  static def gson_version = "2.8.6"
+  // tcnative version from https://github.com/grpc/grpc-java/blob/master/SECURITY.md#netty
+  static def netty_version = "4.1.52.Final"
+  // google-auth-library version from https://search.maven.org/artifact/io.grpc/grpc-auth/1.36.0/jar
+  static def google_auth_version = "0.22.2"
+  static def proto_google_common_protos_version = "2.0.1"
+  static def opencensus_version = "0.28.0"
+  static def perfmark_version = "0.23.0"
+  static def lzma_java_version = "1.3"
+  static def protobuf_javanano_version = "3.0.0-alpha-5"
+  static def jzlib_version = "1.1.3"
+  static def compress_lzf_version = "1.0.3"
+  static def lz4_version = "1.3.0"
+  static def bouncycastle_version = "1.54"
+  static def conscrypt_version = "2.5.1"
+  static def alpn_api_version = "1.1.2.v20150522"
+  static def npn_api_version = "1.1.1.v20141010"
+  static def jboss_marshalling_version = "1.4.11.Final"
+  static def jboss_modules_version = "1.1.0.Beta1"
+
+  /** Returns the list of compile time dependencies. */
+  static List<String> dependencies() {
+    return [
+      "com.google.guava:guava:$guava_version",
+      "com.google.protobuf:protobuf-java:$protobuf_version",
+      "com.google.protobuf:protobuf-java-util:$protobuf_version",
+      "com.google.code.gson:gson:$gson_version",
+      "io.grpc:grpc-auth:$grpc_version",
+      "io.grpc:grpc-core:$grpc_version",
+      "io.grpc:grpc-context:$grpc_version",
+      "io.grpc:grpc-netty:$grpc_version",
+      "io.grpc:grpc-protobuf:$grpc_version",
+      "io.grpc:grpc-stub:$grpc_version",
+      "io.netty:netty-transport-native-epoll:$netty_version",
+      // tcnative version from https://github.com/grpc/grpc-java/blob/master/SECURITY.md#netty
+      "io.netty:netty-tcnative-boringssl-static:2.0.34.Final",
+      "com.google.auth:google-auth-library-credentials:$google_auth_version",
+      "io.grpc:grpc-testing:$grpc_version",
+      "com.google.api.grpc:proto-google-common-protos:$proto_google_common_protos_version",
+      "io.opencensus:opencensus-api:$opencensus_version",
+      "io.opencensus:opencensus-contrib-grpc-metrics:$opencensus_version",
+      "io.perfmark:perfmark-api:$perfmark_version",
+      "com.github.jponge:lzma-java:$lzma_java_version",
+      "com.google.protobuf.nano:protobuf-javanano:$protobuf_javanano_version",
+      "com.jcraft:jzlib:$jzlib_version",
+      "com.ning:compress-lzf:$compress_lzf_version",
+      "net.jpountz.lz4:lz4:$lz4_version",
+      "org.bouncycastle:bcpkix-jdk15on:$bouncycastle_version",
+      "org.bouncycastle:bcprov-jdk15on:$bouncycastle_version",
+      "org.eclipse.jetty.alpn:alpn-api:$alpn_api_version",
+      "org.eclipse.jetty.npn:npn-api:$npn_api_version",
+      "org.jboss.marshalling:jboss-marshalling:$jboss_marshalling_version",
+      "org.jboss.modules:jboss-modules:$jboss_modules_version"
```

Comment:
If we don't have these dependencies, checkJavaLinkage task shows the following errors. But as long as we don't use jboss marshall or alpn SSL engine, we don't need them.

```
Class org.eclipse.jetty.npn.NextProtoNego is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.JettyNpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.conscrypt.Conscrypt is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.ConscryptAlpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.eclipse.jetty.npn.NextProtoNego$ServerProvider is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.JettyNpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.eclipse.jetty.alpn.ALPN$ServerProvider is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.JettyAlpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.eclipse.jetty.alpn.ALPN is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.JettyAlpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.conscrypt.BufferAllocator is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.ConscryptAlpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.eclipse.jetty.alpn.ALPN$ClientProvider is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.JettyAlpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.conscrypt.HandshakeListener is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.ConscryptAlpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.eclipse.jetty.npn.NextProtoNego$ClientProvider is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.ssl.JettyNpnSslEngine (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.jboss.marshalling.Marshaller is not found;
  referenced by 3 class files
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ThreadLocalMarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.MarshallingEncoder (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.CompatibleMarshallingEncoder (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.jboss.marshalling.MarshallerFactory is not found;
  referenced by 4 class files
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ThreadLocalMarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ThreadLocalUnmarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.DefaultMarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.DefaultUnmarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.jboss.marshalling.Unmarshaller is not found;
  referenced by 4 class files
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ThreadLocalUnmarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ContextBoundUnmarshallerProvider (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.MarshallingDecoder (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.CompatibleMarshallingDecoder (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.jboss.marshalling.ByteInput is not found;
  referenced by 2 class files
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.LimitingByteInput (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ChannelBufferByteInput (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
Class org.jboss.marshalling.ByteOutput is not found;
  referenced by 1 class file
    org.apache.beam.vendor.grpc.v1p36p0.io.netty.handler.codec.marshalling.ChannelBufferByteOutput (org.apache.beam:beam-vendor-grpc-1_36_0:0.1)
  Cause:
    Unknown
```

How can Linkage Checker help here?

https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1976

Reply:
I excluded a bunch of classes that were wrappers for optional libraries we didn't use. You might try the same thing:
https://github.com/apache/beam/pull/14146/files#diff-7a309e4c96aaa00cb19271bf8e3df2979434bb29dd426472bb547c27ee6df1a9R76

Reply:
@apilloud Thank you very much. That sounds a correct way to handle this. I'll try to remove the unused classes that cause the reference errors.